Mathlib/Geometry/Manifold/ContMDiff/Defs.lean

```diff
@@ -619,6 +619,30 @@ theorem contMDiffWithinAt_iff_nat {n : ℕ∞} :
   · exact contMDiffWithinAt_infty.2 fun n => h n le_top
   · exact h n le_rfl
 
+theorem contMDiffAt_iff_nat {n : ℕ∞} :
+    ContMDiffAt I I' n f x ↔ ∀ m : ℕ, (m : ℕ∞) ≤ n → ContMDiffAt I I' m f x := by
+  simp [← contMDiffWithinAt_univ, contMDiffWithinAt_iff_nat]
+
+/-- A function is `C^n` within a set at a point iff it is `C^m` within this set at this point, for
+any `m ≤ n` which is different from `∞`. The interest of this result is that, when `m ≠ ∞`, being
```

Comment:
This may be a matter of taste: I'd write "This result is useful because" instead of "The interest of this result is that,"
(The result is not a sentient being, at least in my philosophical worldview.)